srml/slashing/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+#![deny(missing_docs, rust_2018_idioms)]
+
+//! Slashing interface
+//!
+//! That gives functionality to specialize slashing and misconduct for a given type
+//! In order to customize severity level and misconduct fees.
+//!
+
+// mod types;
+#[cfg(test)]
+mod tests;
+
+use parity_codec::Codec;
+use primitives::traits::{SimpleArithmetic, MaybeSerializeDebug};
+use srml_support::traits::{Currency, LockableCurrency};
+
+type BalanceOf<T> = <<T as Misconduct>::Currency as Currency<<T as system::Trait>::AccountId>>::Balance;
+
+/// Estimates severity level based on misconduct
+pub trait Misconduct: system::Trait {
```

Comment:
I am not the biggest fan of this myself but afaik all modules have a local trait called `Trait`.

```suggestion
pub trait Trait: system::Trait {
```

Reply:
(given that it is a module. After reading a bit down it looks like it is not?)